scripts/setup-skel.py

```diff
@@ -0,0 +1,174 @@
+    remove_items(
+        directory=dest_dir,
+        items=[
+            "setup",
+            ".Rproj.user",
+            ".Rhistory",
+            ".gitignore",
+        ],
+    )
+
+    # remove all files with the .nb.html extension
+    for file in dest_dir.glob("*.nb.html"):
+        file.unlink()
+    if as_reference:
+        # remove all files with the -live.Rmd extension
+        for file in dest_dir.glob("*-live.Rmd"):
+            file.unlink()
+    else:
+        # remove Rmd files *if there is a matching -live version*
+        for file in dest_dir.glob("*.Rmd"):
+            if (dest_dir / (file.stem + "-live.Rmd")).is_file():
+                file.unlink()
+
+
+def main() -> None:
+    parser = argparse.ArgumentParser(
+        description="Create a skel directory for user setup for CCDL projects"
+    )
+    parser.add_argument(
+        "-b",
+        "--base-dir",
+        type=pathlib.Path,
+        default=pathlib.Path("/etc/skel-templates/training-modules"),
+        help="Directory containing the modules to be included in the training",
+    )
+    parser.add_argument(
+        "-s",
+        "--skel-dir",
+        type=pathlib.Path,
+        help="Directory to be used as the output skel directory. This will be created if it does not exist",
+    )
+    parser.add_argument(
+        "-m",
+        "--module-file",
+        type=pathlib.Path,
+        help="A JSON file containing the list of modules to be included in the training",
+    )
+    args = parser.parse_args()
+
+    if not args.base_dir.is_dir():
+        exit(f"Base directory {args.base_dir} does not exist or is not a directory")
+    if not args.module_file.is_file() and not args.module_file.suffix == ".json":
+        exit("A JSON module file must be provided.")
+
+    module_data = json.loads(args.module_file.read_text())
+    modules = module_data.get("modules", [])
+    reference_modules = module_data.get("reference-modules", [])
+
+    if any(m not in ALL_MODULES for m in modules):
+        exit(f"Invalid module(s) specified. Available modules: {ALL_MODULES}")
+    if any(m not in ALL_MODULES for m in reference_modules):
+        exit(f"Invalid reference module(s) specified. Available modules: {ALL_MODULES}")
+
+    # check that modules are in the base directory
+    if any(not (args.base_dir / m).is_dir() for m in modules):
+        exit(f"One or more modules do not exist in the base directory {args.base_dir}")
+
+    # run the link-data.sh script in the base directory to create required symlinks
+    link_script = args.base_dir / "scripts/link-data.sh"
+    if not link_script.is_file():
+        exit(f"The required script {link_script} does not exist or is not a file")
+    subprocess.run(["bash", link_script], check=True)
+
+    # create the target directory
+    args.skel_dir.mkdir(parents=True, exist_ok=True)
+    # create the shared-data link
+    (args.skel_dir / "shared-data").symlink_to("/shared/data")
```

Comment:
General question, but I believe this is what is causing me problems – how do you envision folks testing this locally?

Reply:
What problem are you seeing? I would expect this to create a broken symlink when run locally, but not "fail".

As far as testing: I would expect most testing to happen on the server where the `/shared` directories exist.

Reply:
The problem was the _rerunning_ of the script after removing some local files, so I think we're good.